third_party/protobuf/protobuf-3.1.0.patch

```diff
@@ -0,0 +1,29 @@
+diff -Nrup protobuf-3.1.0.orig/src/google/protobuf/io/zero_copy_stream_impl_lite.h protobuf-3.1.0/src/google/protobuf/io/zero_copy_stream_impl_lite.h
```

Comment:
This patch already exists for 2.5.0, I just regenerate it for 3.1.0. The reasons why we need it are:

1) The C++11 version is faster, it saves an if check on super hot path.
2) We compile our stream manager with C++11, so it is safe to just remove the macro
3) I don't see any configure parameter to enable it, so we have to patch the source directly.

Not sure if it is really worth a README, since we don't change the patch often. It is generated manually actually.